hbase-common/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfo.java

```diff
@@ -114,6 +121,34 @@ public boolean removeServer(Address hostPort) {
     return servers.remove(hostPort);
   }
 
+  /**
+   * Getter for fetching an unmodifiable {@link #configuration} map.
+   */
+  public Map<String, String> getConfiguration() {
+    // shallow pointer copy
+    return Collections.unmodifiableMap(configuration);
+  }
+
+  /**
+   * Setter for storing a configuration setting in {@link #configuration} map.
+   * @param key Config key.
+   * @param value String value. If null, removes the setting.
+   */
+  public void setConfiguration(String key, String value) {
```

Comment:
We have a removeConfiguration so let's not allow null value here?

Reply:
OK

Reply:
The implementation still accepts null value?

Reply:
Has thrown a exception when accepts null value, it isn't right? Also add a `@NonNull` annotation?

Reply:
Oh, you can just use Objects.requireNonNull. I see there is still a null check so do not look closely the line inside the if...

Reply:
Right, replaced by Objects.requireNonNull.